charts/consul/templates/ingress-gateways-deployment.yaml

```diff
@@ -107,374 +146,172 @@ spec:
+          -gateway-kind="ingress-gateway" \
+          -consul-node-name="k8s-service-mesh" \
+          -proxy-id-file=/consul/service/proxy-id \
+          -service-name={{ template "consul.fullname" $root }}-{{ .name }} \
+          -log-level={{ default $root.Values.global.logLevel }} \
+          -log-json={{ $root.Values.global.logJSON }}
+        volumeMounts:
         - name: consul-service
-          emptyDir:
-            medium: "Memory"
+          mountPath: /consul/service
         {{- if $root.Values.global.tls.enabled }}
-        {{- if not (and $root.Values.externalServers.enabled $root.Values.externalServers.useSystemRoots) }}
         - name: consul-ca-cert
-          secret:
-            {{- if $root.Values.global.tls.caCert.secretName }}
-            secretName: {{ $root.Values.global.tls.caCert.secretName }}
-            {{- else }}
-            secretName: {{ template "consul.fullname" $root }}-ca-cert
-            {{- end }}
-            items:
-            - key: {{ default "tls.crt" $root.Values.global.tls.caCert.secretKey }}
-              path: tls.crt
-        {{- end }}
-        {{- if $root.Values.global.tls.enableAutoEncrypt }}
-        - name: consul-auto-encrypt-ca-cert
-          emptyDir:
-            medium: "Memory"
+          mountPath: /consul/tls/ca
+          readOnly: true
         {{- end }}
+        resources:
+          requests:
+            memory: "50Mi"
+            cpu: "50m"
+          limits:
+            memory: "50Mi"
+            cpu: "50m"
+      containers:
+      - name: ingress-gateway
+        image: {{ $root.Values.global.imageConsulDataplane | quote }}
+        {{- if (default $defaults.resources .resources) }}
+        resources: {{ toYaml (default $defaults.resources .resources) | nindent 10 }}
         {{- end }}
-      initContainers:
-        # We use the Envoy image as our base image so we use an init container to
-        # copy the Consul binary to a shared directory that can be used when
-        # starting Envoy.
-        - name: copy-consul-bin
-          image: {{ $root.Values.global.image | quote }}
-          command:
-          - cp
-          - /bin/consul
-          - /consul-bin/consul
-          volumeMounts:
-          - name: consul-bin
-            mountPath: /consul-bin
-          {{- $initContainer := .initCopyConsulContainer }}
-          {{- if (or $initContainer $defaults.initCopyConsulContainer) }}
-          {{- if (default $defaults.initCopyConsulContainer.resources $initContainer.resources) }}
-          resources: {{ toYaml (default $defaults.initCopyConsulContainer.resources $initContainer.resources) | nindent 12 }}
-          {{- end }}
-          {{- end }}
-        {{- if (and $root.Values.global.tls.enabled $root.Values.global.tls.enableAutoEncrypt) }}
-        {{- include "consul.getAutoEncryptClientCA" $root | nindent 8 }}
+        volumeMounts:
+        {{- if $root.Values.global.tls.enabled }}
```

Comment:
this doesn't need to be mounted if external servers with use system roots is set